packages/odyssey-react/src/components/TextInput/TextInput.tsx

```diff
@@ -219,6 +239,16 @@ export const TextInput = withTheme(
               {suffix}
             </span>
           )}
+          {isSearchTextInput && showClearButton && (
```

Comment:
Put all this and the suffix section in a separate component? Not sure how that works with the `withTheme`

Reply:
Similar to the CSS above, if we had the right Button variant, could Button populate `suffix` instead?

Reply:
That's what I was referring to in my comment about having another component, maybe something like `EndDecoration` that would encapsulate the span/suffix/button. 
Do you think that is preferred. Would it live in `Button` or in `TextInput` and how would I control visibility of the Component when the root is hovered/focused?
Right now I do it via css
```
.root:hover .clear,
.root:focus-within .clear {
  display: block;
}
```
But how would that work if everything is abstracted out to a different component? Maybe it'll just work and I'm over complicating it?